source/fundamentals/data-formats/documents.txt

```diff
@@ -284,6 +289,94 @@ For more information on retrieving and manipulating MongoDB data, see our
+You can customize the format of JSON in ``JsonObject`` by specifying a
+:java-docs:`JsonObjectCodec</apidocs/bson/org/bson/codecs/JsonObjectCodec.html>`
+and passing it a
+:java-docs:`JsonWriterSettings</apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+object. For more information on JSON formats, see
+our :doc:`Extended JSON guide </fundamentals/data-formats/document-data-format-extended-json/>`.
+For more information on specifying codecs, see our 
+:doc:`Codecs guide </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate a sample ``JsonObject`` 
+instance wrapping an Extended JSON string containing different types of key value pairs:
+
+.. code-block:: java
+
+   String ejsonStr =  "{"
+                     +" \"_id\": {"
+                     +"  \"$oid\": \"6035210f35bd203721c3eab8\""
+                     +" },"
+                     +" \"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\","
+                     +" \"dateOfDeath\": {"
+                     +"  \"$date\": \"2014-04-17T04:00:00Z\""
+                     +" },"
+                     +" \"novels\": ["
+                     +"  {"
+                     +"   \"title\": \"One Hundred Years of Solitude\","
+                     +"   \"yearPublished\": 1967"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Chronicle of a Death Foretold\","
+                     +"   \"yearPublished\": 1981"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Love in the Time of Cholera\","
+                     +"   \"yearPublished\": 1985"
+                     +"  }"
+                     +" ]"
+                     +"}";
+   JsonObject author = new JsonObject(ejsonStr);
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``JsonObject`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<JsonObject> collection = database.getCollection("authors", JsonObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample JSON data from the 
+collection using the following code:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   JsonObject query = new JsonObject("{\"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\"}");
+   JsonObject json = collection.find(query).first();
+   if (json != null) {
+       //this is equivalent to System.out.println("raw extended json string: " + json);
```

Comment:
1) Formatting is off for a comment, it should contain a space after the "//" I don't think we have identified a specific Java code style guide we want to strictly follow, but I think the default Eclipse or IntelliJ ones will add the space.
2) Thanks for including information about this, but I'm not sure this makes sense. First, I'm not sure if you meant "raw" since it's just an extended json string. You might refer to a "raw" format as something that has some binary or unreadable data that needs conversion to read. Let's just remove this line. If you still want to show the user the equivalency between toJson and toString, you can add that in a "..note::" directive.

Reply:
Thanks for the feedback. Comment deleted and print statment updated here: https://github.com/mongodb/docs-java/pull/57/commits/1d26176eac8a652261f7cd1511c6e3ba9430f451

output updated here: https://github.com/mongodb/docs-java/pull/57/commits/26577c6da0c2e554f62a4410286f763c921ae80d

I don't think informing users about the equivalence of `toString` and `toJson` provides a lot of value, so I am personally okay with leaving it out.